observability/test/run_o11y_tests.py

```diff
@@ -1035,6 +1047,47 @@ def test_logging_connect_trace(self) -> None:
             trace_id = entry.trace.split('/')[-1]
             self.assertTrue(trace_id in trace_ids)
 
+    def _test_trace_diff_setting_endpoints(self,
+                                           server_trace_config: Optional[Dict[str, Any]],
+                                           client_trace_config: Optional[Dict[str, Any]]) -> None:
+        self.initialize_config()
+        self.enable_server_trace(server_trace_config)
+        self.enable_client_trace(client_trace_config)
+        self.setup_and_run_rpc([InteropAction('large_unary', num_times = 10)])
+
+    def test_trace_diff_setting_endpoints(self) -> None:
+        self._test_trace_diff_setting_endpoints(server_trace_config = {
+            'sampling_rate': 0.00
+        }, client_trace_config = {
+            'sampling_rate': 0.00
+        })
+        trace_results = CloudTraceInterface.query_traces_from_cloud(self)
+        trace_results.test_traces_count(0)
+
+        self._test_trace_diff_setting_endpoints(server_trace_config = {
+            'sampling_rate': 0.00
+        }, client_trace_config = {
+            'sampling_rate': 1.00
+        })
+        trace_results = CloudTraceInterface.query_traces_from_cloud(self)
+        trace_results.test_traces_count(10)
+        if not trace_results.has_recv_span():
+            self.fail('Should have some Recv span')
+        if not trace_results.has_sent_span():
+            self.fail('Should have some Sent span')
+
+        self._test_trace_diff_setting_endpoints(server_trace_config = {
+            'sampling_rate': 1.00
+        }, client_trace_config = {
+            'sampling_rate': 0.00
+        })
```

Comment:
Can we also add test to set probabilities between (0.0, 1.0) fro server config and verify if server honors the sampling rate? 
If you have a similar test case, feel free to ignore.

Reply:
We already have such a test case https://github.com/GoogleCloudPlatform/grpc-gcp-tools/blob/master/observability/test/run_o11y_tests.py#L882-L893